integration-test/run.sh

```diff
@@ -0,0 +1,179 @@
+#!/bin/bash
+set -ex
+
+source extra.sh
+
+echo "Required variable check"
+if [ -z "$CONTAINER_PREFIX" ]; then
+  echo "You must set the shell env variable CONTAINER_PREFIX so we can push the containers to the correct place"
+  exit 1
+fi
+# Sort of required, we can set is kubectl cluster-info works
+if [ -z "$K8S_ENDPOINT" ]; then
+  echo "You should configure your K8S_ENDPOINT attempting cluster-info"
+  K8S_ENDPOINT=$(kubectl cluster-info)
+fi
+
+echo "Required command check"
+if ! command -v helm &> /dev/null
+then
+  echo "Helm not found, please install helm https://helm.sh/docs/intro/install/"
+  exit 1
+fi
+echo "Configuring command for downloads"
+DL="axel"
+if ! command -v axel &> /dev/null
+then
+  echo "Falling back to wget from axel, try brew install axel or apt-get install axel for faster downloads."
+  DL="wget"
+fi
+
+
+echo "Configuring default variables if unset"
+INTEGRATION_RUN_DIR=${INTEGRATION_RUN_DIR:-$(mktemp -d -t iceberg-int-XXXXXXXXXX)}
+# Repos and branches to test
+SPARK_REPO=${SPARK_REPO:-"https://github.com/holdenk/spark"}
+SPARK_BRANCHES=${SPARK_BRANCHES:-"master future"}
+SPARK_BRANCHES_ARRAY=($(echo ${SPARK_BRANCHES} | tr " " "\n"))
+SPARK_VERSION=${SPARK_VERSION:-3.0.1}
+SPARK_SUBDIR=${SPARK_SUBDIR:-spark-${SPARK_VERSION}-bin-hadoop3.2}
+SPARK_HOME=${SPARK_HOME:-"${INTEGRATION_RUN_DIR}/${SPARK_SUBDIR}"}
+SPARK_ARCHIVE=${SPARK_ARCHIVE:-${SPARK_SUBDIR}.tgz}
+FLINK_VERSION=${FLINK_VERSION:-1.11}
+# Storage
+MINIO_REPO=${MINIO_REPO:-minio/minio}
+MINIO_TAG=${MINIO_TAG:-RELEASE.2021-02-07T01-31-02Z}
+MINIO_MC_REPO=${MINIO_MC_REPO:-minio/mc}
+MINIO_MC_TAG=${MINIO_MC_TAG:-RELEASE.2021-02-07T02-02-05Z}
+# K8s
+if [ -z "${TEST_NS}" ]; then
+  export TEST_NS=iceberg-integration
+  kubectl create namespace "${TEST_NS}"
+fi
+# Apply any extra configuration (e.g. networking, etc.)
+if [ -f extra.yaml ]; then
+  kubectl apply -f extra.yaml
+fi
+if [ -z "$SERVICE_ACCOUNT" ]; then
+  echo "No service account configured, making one"
+  kubectl get serviceaccount spark-iceberg || kubectl create serviceaccount spark-iceberg --namespace ${TEST_NS}
+  kubectl get rolebinding spark-ice-role || kubectl create rolebinding spark-ice-role --role=edit --serviceaccount=${TEST_NS}:spark --namespace=${TEST_NS}
+  export SERVICE_ACCOUNT=spark
+fi
+TAG=$(date +%s)
+# Archs to build for, e.g. "--platform linux/amd64,linux/arm64"
+#ARCHS=${ARCHS:""}
+
+#TODO: Add multi-arch
+#ARCHS=${ARCHS:-"--platform linux/amd64,linux/arm64"}
+
+echo "Initial setup"
+INTEGRATION_DIR=$(dirname "$0")
+cd "${INTEGRATION_DIR}"
+INTEGRATION_DIR=$(pwd)
+cd ..
+ICEBERG_DIR=$(pwd)
+#./gradlew build
```

Comment:
Probably we should add a `-x test` (and not forgetting that `-x integTest` exists, depending on how we set this up). 

Happy to make these changes, just didn't want to step all over the branch in the middle of the night on you 👍

Reply:
Oh yeah go for it, please feel free to make the changes.